torch/amp/autocast_mode.py

```diff
@@ -257,11 +257,11 @@ def __init__(
             self._cache_enabled = cache_enabled
 
         if self.device == "cpu":
-            supported_dtype = [torch.bfloat16]
+            supported_dtype = [torch.bfloat16, torch.float16]
             if self.fast_dtype not in supported_dtype and enabled:
                 error_message = "In CPU autocast, but the target dtype is not supported. Disabling autocast.\n"
                 error_message += (
-                    "CPU Autocast only supports dtype of torch.bfloat16 currently."
+                    "CPU Autocast only supports dtype of torch.bfloat16 and torch.float16 currently."
```

Comment:
Hmm, I think we need to be careful about which ops are supported for float16 autocasting on CPU. for GPU, bfloat16 autocasts are different from float16.

Reply:
See https://pytorch.org/docs/stable/amp.html#cuda-ops-that-can-autocast-to-float16

Reply:
AFAIK, bfloat16 and float16 autocast on GPU share the same cast policy, no? For CPU, we are going to share the same list for both data types too.

Reply:
Both CUDA and CPU share the same list for BF16 and FP16 data types for autocast.